### PR TITLE
Bug 1426694 added section on sending logs to external rsyslog

### PR DESCRIPTION
**Pull request by bfallonf**
For: https://bugzilla.redhat.com/show_bug.cgi?id=1426694

and: #5415

cc: @sqtran

### CODE REVIEW

File: install_config/aggregate_logging.adoc [hunk 1]
```diff
@@ -1462,6 +1462,149 @@ have the same multi-tenant capabilities and your data will not be restricted by
 user access to a particular project.
 ====
 
+[[sending-logs-to-external-rsyslog]]
+== Sending Logs to an External Rsyslog Instance
+
+Use the `fluent-plugin-remote-syslog` plug-in on the host to send logs to an external rsyslog server.
```
Comment (bfallonf) [inline install_config/aggregate_logging.adoc]: The plugin is references "syslog", but am I ok referencing rsyslog here?

File: install_config/aggregate_logging.adoc [hunk 2]
```diff
@@ -1462,6 +1462,149 @@ have the same multi-tenant capabilities and your data will not be restricted by
...
+
+[WARNING]
+====
+This implementation is insecure, and should only be used in environments
+where you can guarantee no snooping on the connection.
+====
+
+[cols="3,7",options="header"]
+|===
+|Parameter |Description
```
Comment (bfallonf) [inline install_config/aggregate_logging.adoc]: I'm not sure what these parameters are for. How are they used?

Comment (sqtran) [reply install_config/aggregate_logging.adoc]: These are the configuration parameters for the gem, exposed via ENV variables.  They're one-to-one with what's described here too.  https://github.com/jkupferer/fluent-plugin-remote-syslog

Take a look at this too https://github.com/sqtran/origin-aggregated-logging/blob/master/docs/remote-syslog.md  Let me know if this helps answer any of your questions.  Definitely ping me again if it doesn't so I can put together more info.

File: install_config/aggregate_logging.adoc (hunk 1 above)
Comment (sqtran) [reply install_config/aggregate_logging.adoc]: I think they're both compatible, but I was saying "syslog" to keep it in line with what the gem said.

**Comment by bfallonf** [comment]
Thanks @sqtran . I'll put this on to QA.

File: install_config/aggregate_logging.adoc [hunk 3]
```diff
@@ -1468,6 +1468,158 @@ have the same multi-tenant capabilities and your data will not be restricted by
...
+
+Alternatively, you can configure your own custom `fluent.conf` in the
+`logging-fluentd` or `logging-mux` ConfigMaps.
+
+[NOTE]
+====
+Any logs on the external rsyslog server will be raw journald output, and not
+formatted similar to the OpenShift Fluentd output. This means it will not be
+corrolatable and will have no OpenShift or Kubernetes metadata.
+====
```
Comment (richm) [inline install_config/aggregate_logging.adoc]: This is not true when using the fluentd syslog plugin.  When using this plugin, you get the records in the same format that is used to store them in the aggregated logging Elasticsearch, fully annotated with Kubernetes metadata.

This is only true if you use rsyslog to gather the logs, which is a completely different topic.

Comment (bfallonf) [reply install_config/aggregate_logging.adoc]: Sure. Thanks @richm . I deleted the note.

**Comment by bfallonf** [comment]
[rev_history]
|xref:../install_config/aggregate_logging.adoc#install-config-aggregate-logging[Aggregating Container Logs]
|Added the xref:../install_config/aggregate_logging.adoc#sending-logs-to-external-rsyslog[Sending Logs to an External Syslog Server] section.
%

**Comment by bfallonf** [comment]
@openshift/team-documentation PTAL

File: install_config/aggregate_logging.adoc [hunk 4]
```diff
@@ -1468,6 +1468,151 @@ have the same multi-tenant capabilities and your data will not be restricted by
...
+- name: REMOTE_SYSLOG_HOST <1>
+  value: host1
+- name: REMOTE_SYSLOG_HOST_BACKUP
+  value: host2
+- name: REMOTE_SYSLOG_PORT_BACKUP
+  value: 5555
+----
+<1> The desired remote syslog host. Required for each host.
+
+The above will build two destinations. The syslog server on `host1` will be
```
Comment (gaurav-nelson) [inline install_config/aggregate_logging.adoc]: s/The above will build/This builds

File: install_config/aggregate_logging.adoc [hunk 5]
```diff
@@ -1468,6 +1468,151 @@ have the same multi-tenant capabilities and your data will not be restricted by
...
+receiving messages on the default port of `514`, while `host2` will be receiving
+the same messages on port `5555`.
+
+Alternatively, you can configure your own custom `fluent.conf` in the
```
Comment (gaurav-nelson) [inline install_config/aggregate_logging.adoc]: Should 
```
`fluent.conf`
```
be
```
*_fluent.conf_*
?
```

**Comment by gaurav-nelson** [comment]
Line 64: Link to sizing guideline should be `xref:../install_config/aggregate_logging_sizing.adoc#install-config-aggregate-logging-sizing`

**Comment by gaurav-nelson** [comment]
@bfallonf few comments, LGTM otherwise. :national_park:

**Comment by bfallonf** [comment]
Thanks @gaurav-nelson . Suggestions fixed. I'll merge.

**Comment by vikram-redhat** [comment]
@bfallonf - can you check if this needs to go into the 3.9 branch as well?

**Comment by bfallonf** [comment]
@vikram-redhat Not sure what you mean. This has been cherry-picked into the 3.9 branch. Perhaps I didn't add the label. Adding now.

**Pull request merged**